### ISSUE TITLE
Application objects scramble labels during resizing

### ISSUE THREAD

**Issue by ngreenwald** [vanvalenlab/deepcell-tf#369]
**Describe the bug**
The application objects resize the post-processed labels. The resize function uses linear interpolation, which smoothly averages adjacent values, resulting in a gradient which destroys the label. 

Nearest-neighbor interpolation maintains integer values. I opened https://github.com/vanvalenlab/deepcell-toolbox/pull/24 to address this issue and switch to nearest-neighbor interpolation, rather than linear, for labels. However, when downsizing a label output, it maintains integer values but interpolates integer values between adjacent labels, introducing new labels not in the original image.

**Additional context**
I think the best way to handle this is to resize the network output prior to relabeling. That way the interpolation happens on the pixel probabilities, and we can then post-process those resized probabilities. 

However, resize currently expects an array of values, rather than a list. If we concatenate the network output into an array and then pass it to resize, the deep_watershed post-processing function will break, since it expects a list of values. 

TLDR:
Does it make more sense to change resize to expect a list, change the post-processing function to expect an array, or transform the format between each step to maintain the current input arguments? @willgraf @msschwartz21

**Comment by msschwartz21** [comment]
Originally I decided to run resize after post-processing in case the post-processing function had parameters that were size specific. For example, a threshold for removing small objects could cause problems if the resolution of the data is substantially different from the training resolution.

If we can improve the resize function to tolerate labeled images, I think that would be the best course of action.

**Comment by ngreenwald** [comment]
I don't see a way to avoid resize interpolating labels that don't exist
when downsizing. Open to suggestions!

**Comment by willgraf** [comment]
Has this been resolved with vanvalenlab/deepcell-toolbox#24 ?

**Comment by ngreenwald** [comment]
Yes, that solves the issue with the resize function itself. Once we have a new release of the toolbox we'll be able to import the version that has the fix and it'll be fixed on the deepcell-tf side as well
